engine/chain/internal/query/query_service_test.go

```diff
@@ -0,0 +1,131 @@
+package query
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryService(t *testing.T) {
+	logger := zerolog.New(zerolog.NewConsoleWriter(zerolog.ConsoleTestWriter(t)))
+	deepSuffix, _ := uuid.NewUUID()
+	deploymentID := "spv-wallet-" + deepSuffix.String()
+
+	testCases := map[string]struct {
+		txID         string
+		arcToken     string
+		arcURL       string
+		expectErr    error
+		expectNil    bool
+		expectTxID   string
+		expectStatus string
+		applyTimeout timeoutDst
+	}{
+		"Query for MINED transaction": {
+			txID:         minedTxID,
+			arcToken:     arcToken,
+			arcURL:       arcURL,
+			expectErr:    nil,
+			expectTxID:   minedTxID,
+			expectStatus: "MINED",
+		},
+		"Query for unknown transaction": {
+			txID:      unknownTxID,
+			arcToken:  arcToken,
+			arcURL:    arcURL,
+			expectErr: nil,
+			expectNil: true,
+		},
+		"Query for invalid transaction": {
+			txID:      "invalid",
+			arcToken:  arcToken,
+			arcURL:    arcURL,
+			expectErr: spverrors.ErrInvalidTransactionID,
+			expectNil: true,
+		},
+		"Query with wrong token": {
+			txID:      minedTxID,
+			arcToken:  "wrong-token",
+			arcURL:    arcURL,
+			expectErr: spverrors.ErrARCUnauthorized,
+			expectNil: true,
+		},
+		"Query 404 endpoint but reachable": {
+			txID:      minedTxID,
+			arcToken:  arcToken,
+			arcURL:    arcURL + wrongButReachable,
+			expectErr: spverrors.ErrARCUnreachable,
+			expectNil: true,
+		},
+		"Query 404 endpoint with wrong arcURL": {
+			txID:      minedTxID,
+			arcToken:  arcToken,
+			arcURL:    "wrong-url",
+			expectErr: spverrors.ErrARCUnreachable,
+			expectNil: true,
+		},
+		"Query interrupted by ctx timeout": {
+			txID:         minedTxID,
+			arcToken:     arcToken,
+			arcURL:       arcURL,
+			expectErr:    spverrors.ErrARCUnreachable,
+			expectNil:    true,
+			applyTimeout: applyTimeoutCtx,
+		},
+		"Query interrupted by resty timeout": {
+			txID:         minedTxID,
+			arcToken:     arcToken,
+			arcURL:       arcURL,
+			expectErr:    spverrors.ErrARCUnreachable,
+			expectNil:    true,
+			applyTimeout: applyTimeoutResty,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			httpClient, reset := arcMockActivate()
+			defer reset()
+
+			service := NewQueryService(logger, httpClient, tc.arcURL, tc.arcToken, deploymentID)
+
+			ctx := context.Background()
+			if tc.applyTimeout == applyTimeoutCtx {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, 1*time.Millisecond)
+				defer cancel()
+			} else if tc.applyTimeout == applyTimeoutResty {
+				service.httpClient.SetTimeout(1 * time.Millisecond)
+			}
+
+			txInfo, err := service.Query(ctx, tc.txID)
+
+			if tc.expectErr != nil {
```

Comment:
To be honest I've initially implemented it without test-table. But I've forced myself to align with your suggestion to use test-table. 

Splitting into `success cases` and `error cases` won't solve the problem, because for `error cases` I would have to define `timeout errors cases` and `other error cases`.

Taking above into account I would go back to standard test cases without test-table. Agree @dorzepowski ?

Reply:
it depends,
if you have single positive case, then feel free to keep it as standard test
if you have single error case then feel free to keep it as single standard test
if you have single timeout case then feel free to keep it as single standard test

But if you have several test cases, that actually are exactly the same, just input and output is changing, then it's better to have table tests.
IMHO we should pick a tool that the best fits our needs, I don't see any reason to create table tests for single test case.

Reply:
I've reorganized things - check it out 👍